store/generated_key/memstore/v2/memstore.go

```diff
@@ -0,0 +1,219 @@
+package memstore
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/ephemeral_db"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+
+	cert_verifier_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifier"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+const (
+	BytesPerFieldElement = 32
+)
+
+// randomBytes ... Generates random byte slice provided
+// size. Errors when generating are ignored since this is only
+// used for constructing dummy certificates when testing insecure integrations.
+// in the worst case it doesn't work and returns empty arrays which would only
+// impact memstore operation in the event that two identical payloads are provided
+// since they'd resolve to the same commitment and blob key. This shouldn't matter
+// given this is typically used for testing standard E2E functionality against a rollup
+// stack which SHOULD never submit an identical batch more than once.
+func randomBytes(size uint) []byte {
+	entropy := make([]byte, size)
+	_, _ = rand.Read(entropy)
+	return entropy
+}
+
+func randInt(max int64) *big.Int {
+	randInt, _ := rand.Int(rand.Reader, big.NewInt(max))
+	return randInt
+}
+
+func randUint32() uint32 {
+	return uint32(randInt(32).Uint64())
+}
+
+/*
+MemStore is a simple in-memory store for blobs which uses an expiration
+time to evict blobs to best emulate the ephemeral nature of blobs dispersed to
+EigenDA V2 operators.
+*/
+type MemStore struct {
+	*ephemeral_db.DB
```

Comment:
```suggestion
        // Stores a mapping of keccak(MarshalJSON(randomlyGeneratedCert)) -> Blob
	*ephemeral_db.DB
```

Reply:
[27bc9f6](https://github.com/Layr-Labs/eigenda-proxy/pull/305/commits/27bc9f6174c1498c61cc79b963ec52cb85b5ee88)

Reply:
update MarshalJSON->RLP since we changed that

Reply:
[12bee99](https://github.com/Layr-Labs/eigenda-proxy/pull/305/commits/12bee9930fef70c0637c6fb860e8d9a62869cc97)